cfg.sample.toml

```diff
@@ -10,6 +10,23 @@ access_token = ""
 app_client_id = ""
 app_client_secret = ""
 
+# A base shared secret; this can be used to generate the per-repo github secret,
+# which is what's used to verify authenticity of github -> Homu webhooks. The
+# per-repo secret can be computed as follows (you'll need to enter this in
+# the repo webhook configuration):
+#
+# $ echo -n "$owner.$name" | openssl sha1 -hmac "$secret"
```

Comment:
Sorry, I overlooked that there's two HMAC uses here. The second (original) usage to compute `computed_hmac` as a signature over the data to compare to `hmac_sig` looks fine to me. The first usage is the one that's suspicious. We want to (deterministically) derive a `repo_secret` from a `base_secret` and additionally contextual info (`$owner.$name`), and this output needs to be a key, since we're passing it to the (second instance of) HMAC as such. This usage matches the intent of KDFs.

I've done some reading (the HKDF IETF spec, the HKDF paper, etc.) and I still think a KDF such as HDFK is preferable here. HKDF in particular uses HMAC as a primitive, but in a two phase scheme with some additional tricks (feedback to minimize correlation, a counter to prevent against cycle shortcuts, etc.). For example, HKDF can have variable sized output, not just whatever the output size of the underlying HMAC is.

Since we're deriving a key from another key, we should use a key derivation function. HKDF is fast and efficient anyways, so we don't lose anything over just HMAC. If we're really worried about performance impacts, we can precompute and/or cache the `repo_secret`s.

I also see after reading the paper that the `$owner.$name` is better suited for the contextual information of HKDF, not the salt.